### PR TITLE
Migrate Live Data to stateflow (#60) and consume it with collectWithLifeCycle (#61)

### PR DESCRIPTION
**Pull request by chege4179**
I've refactored the state in the SessionsViewModel to stateflow from live data and consumed them with collectAsStateWithLifecycle()

### CODE REVIEW

File: presentation/src/main/java/com/android254/presentation/sessions/components/SessionList.kt [hunk 1]
```diff
@@ -45,20 +46,20 @@ fun SessionList(
...
+    if (loading.value) {
         SessionsLoadingSkeleton()
     } else {
         SwipeRefresh(state = swipeRefreshState, onRefresh = {
             viewModel.refreshSessionList()
         }) {
             LazyColumn(modifier = Modifier.fillMaxHeight()) {
                 itemsIndexed(
-                    items = sessions,
+                    items = sessions.value!!,
```
Comment (wangerekaharun) [inline presentation/src/main/java/com/android254/presentation/sessions/components/SessionList.kt:62]: Avoid !!

File: presentation/src/main/java/com/android254/presentation/sessions/view/SessionsViewModel.kt [hunk 2]
```diff
@@ -79,18 +75,16 @@ class SessionsViewModel @Inject constructor(
...
-                    _selectedFilterOptions.value?.toMutableList()?.apply {
-                        removeAt(index)
-                    }
-            }
+        if (_selectedFilterOptions.value.contains(option)) {
+            val index = _selectedFilterOptions.value.indexOf(option)
+            _selectedFilterOptions.value =
+                _selectedFilterOptions.value.toMutableList().apply {
+                    removeAt(index)
+                }!!
```
Comment (wangerekaharun) [inline presentation/src/main/java/com/android254/presentation/sessions/view/SessionsViewModel.kt:83]: Avoid!!

**Review by wangerekaharun — COMMENTED** [review]
Left some comments, also SessionDetailsScreen has observeAsState as well. Generally avoid double bangers

**Comment by yveskalume** [comment]
I also left a comment related to encapsulation

**Comment by chege4179** [comment]
Noted

**Pull request closed without merging**